### PR TITLE
Suggestion box for edit input

### PR DESCRIPTION
**Pull request by perunt**
### Details
This PR updates the suggestion box (in edit mode) to appear right next to the cursor, making it work like inline autosuggestion.

### Fixed Issues

$ https://github.com/Expensify/App/issues/19643
PROPOSAL:

### Tests
Tests
Testing Emoji Suggestions:

1. Open the chat interface.
2. Edit message
3. Type an emoji trigger, starting with :sm followed by at least two more characters (e.g., :smile).
4. Verify that the suggestion box appears adjacent to the cursor.
5. Move the cursor to a new position in the text field.
6. Again, type an emoji-like text with at least two characters following the :sm prefix.
7. Confirm that the suggestion box appears next to the cursor as expected.

Testing Mention Suggestions:
1. Open the chat interface.
2. Edit message
3. Initiate a mention by typing the @ symbol followed by any text (e.g., @username).
4. Ensure the suggestion box pops up next to the cursor.
5. Reposition the cursor within the text field.
6. Type another mention, starting with the @ symbol.
7. Check that the suggestion box is correctly displayed adjacent to the cursor.

- [x] Verify that no errors appear in the JS console

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/48593211/bbf26966-c5e1-4c4b-8324-d02824a3f26c

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/48593211/dc4f4e14-663a-4b51-84c8-fa407b1c5796

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/48593211/c83ca63a-3a57-47d1-88a7-487d60f3daf4

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/48593211/40ffef2a-ad7d-4d5a-ae8d-6aea690b0183

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/48593211/9e5792b3-b011-4a46-8f24-3911da6ea1bd

</details>

### CODE REVIEW

**Comment by perunt** [comment]
I found one issue which is also reproducible for Composer: we do not hide suggestions while scrolling the main chat. I want to address this separately, as after a quick research, I understand that it could involve a refactor of sharing main component scroll info. I don't want to expand and delay this PR @puneetlath  are you okay with that?

**Comment by puneetlath** [comment]
I think addressing separately is fine.

**Comment by perunt** [comment]
hey @rayane-djouah, just friendly reminder 😄

**Review by rayane-d — CHANGES_REQUESTED** [review]

File: src/components/AutoCompleteSuggestions/index.tsx [hunk 1]
```diff
@@ -66,6 +67,8 @@ function AutoCompleteSuggestions<TSuggestion>({measureParentContainerAndReportCu
 
     const suggestionsLength = props.suggestions.length;
 
+    // console.log('emoji_FOCUS', isFocused);
+
```
[the remark this answers is not in the document]
Comment (perunt) [reply src/components/AutoCompleteSuggestions/index.tsx:71]: 👍🏻

File: src/components/AutoCompleteSuggestions/index.tsx [hunk 2]
```diff
@@ -77,15 +80,13 @@ function AutoCompleteSuggestions<TSuggestion>({measureParentContainerAndReportCu
...
-
             const contentMaxHeight = measureHeightOfSuggestionRows(suggestionsLength, true);
             const contentMinHeight = measureHeightOfSuggestionRows(suggestionsLength, false);
-            const isEnoughSpaceAboveForBig = windowHeight - bottomValue - contentMaxHeight > CONST.AUTO_COMPLETE_SUGGESTER.SUGGESTION_BOX_MAX_SAFE_DISTANCE;
-            const isEnoughSpaceAboveForSmall = windowHeight - bottomValue - contentMinHeight > CONST.AUTO_COMPLETE_SUGGESTER.SUGGESTION_BOX_MAX_SAFE_DISTANCE;
+            let bottomValue = windowHeight - (cursorCoordinates.y - scrollValue + y) - keyboardHeight;
+            const widthValue = isSmallScreenWidth ? width : CONST.AUTO_COMPLETE_SUGGESTER.BIG_SCREEN_SUGGESTION_WIDTH;
 
+            const isEnoughSpaceAboveForBig = y + (cursorCoordinates.y - scrollValue) > contentMaxHeight + topInset + CONST.AUTO_COMPLETE_SUGGESTER.SUGGESTION_BOX_MAX_SAFE_DISTANCE;
+            const isEnoughSpaceAboveForSmall = y + (cursorCoordinates.y - scrollValue) > contentMinHeight + topInset + CONST.AUTO_COMPLETE_SUGGESTER.SUGGESTION_BOX_MAX_SAFE_DISTANCE;
```
[the remark this answers is not in the document]
Comment (perunt) [reply src/components/AutoCompleteSuggestions/index.tsx:89]: 👍🏻

File: src/components/AutoCompleteSuggestions/index.tsx [hunk 3]
```diff
@@ -106,18 +107,19 @@ function AutoCompleteSuggestions<TSuggestion>({measureParentContainerAndReportCu
             } else {
                 // calculation for big suggestion box below the cursor
                 measuredHeight = measureHeightOfSuggestionRows(suggestionsLength, true);
-                bottomValue = windowHeight - y - cursorCoordinates.y + scrollValue - measuredHeight - CONST.AUTO_COMPLETE_SUGGESTER.SUGGESTION_ROW_HEIGHT;
+                bottomValue = windowHeight - y - cursorCoordinates.y + scrollValue - measuredHeight - CONST.AUTO_COMPLETE_SUGGESTER.SUGGESTION_ROW_HEIGHT - keyboardHeight - 0;
```
[the remark this answers is not in the document]
Comment (perunt) [reply src/components/AutoCompleteSuggestions/index.tsx:110]: 👍🏻

File: src/pages/home/report/ReportActionItemMessageEdit.tsx [hunk 4]
```diff
@@ -340,15 +374,78 @@ function ReportActionItemMessageEdit(
...
+    const measureContainer = useCallback(
+        (callback: MeasureInWindowOnSuccessCallback) => {
+            if (!containerRef.current) {
+                return;
+            }
+            containerRef.current.measureInWindow(callback);
+        },
+        // We added isComposerFullSize in dependencies so that when this value changes, we recalculate the position of the popup
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        [isFocused],
```
[the remark this answers is not in the document]
Comment (perunt) [reply src/pages/home/report/ReportActionItemMessageEdit.tsx:410]: 👍🏻

File: src/pages/home/report/ReportActionItemMessageEdit.tsx [hunk 5]
```diff
@@ -340,15 +374,78 @@ function ReportActionItemMessageEdit(
                 return;
             }
             const keyEvent = e as KeyboardEvent;
-            if (keyEvent.key === CONST.KEYBOARD_SHORTCUTS.ENTER.shortcutKey && !keyEvent.shiftKey) {
+            const isShortcutEnter = keyEvent.key === CONST.KEYBOARD_SHORTCUTS.ENTER.shortcutKey;
+            const isShortcutEscape = keyEvent.key === CONST.KEYBOARD_SHORTCUTS.ESCAPE.shortcutKey;
```
[the remark this answers is not in the document]
Comment (perunt) [reply src/pages/home/report/ReportActionItemMessageEdit.tsx:378]: What exactly const do you talk about?

File: src/pages/home/report/ReportActionCompose/Suggestions.tsx [hunk 6]
```diff
@@ -124,6 +124,12 @@ function Suggestions(
         suggestionEmojiRef.current?.setShouldBlockSuggestionCalc(shouldBlock);
         suggestionMentionRef.current?.setShouldBlockSuggestionCalc(shouldBlock);
     }, []);
+    const checkIfSuggestionVisible = useCallback((): boolean => {
+        const isEmojiVisible = suggestionEmojiRef.current?.checkIfSuggestionVisible();
+        const isSuggestionVisible = suggestionMentionRef.current?.checkIfSuggestionVisible();
```
[the remark this answers is not in the document]
Comment (perunt) [reply src/pages/home/report/ReportActionCompose/Suggestions.tsx:129]: 👍🏻

File: src/pages/home/report/ReportActionCompose/Suggestions.tsx [hunk 7]
```diff
@@ -124,6 +124,12 @@ function Suggestions(
...
+        // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+        return isEmojiVisible || isSuggestionVisible || false;
```
[the remark this answers is not in the document]
Comment (perunt) [reply src/pages/home/report/ReportActionCompose/Suggestions.tsx:131]: 👍🏻

File: src/components/AutoCompleteSuggestions/index.tsx [hunk 8]
```diff
@@ -77,15 +80,13 @@ function AutoCompleteSuggestions<TSuggestion>({measureParentContainerAndReportCu
...
                     : xCoordinatesOfCursor;
-
-            let bottomValue = windowHeight - y - cursorCoordinates.y + scrollValue - (keyboardHeight || bottomInset);
-            const widthValue = isSmallScreenWidth ? width : CONST.AUTO_COMPLETE_SUGGESTER.BIG_SCREEN_SUGGESTION_WIDTH;
-
             const contentMaxHeight = measureHeightOfSuggestionRows(suggestionsLength, true);
             const contentMinHeight = measureHeightOfSuggestionRows(suggestionsLength, false);
-            const isEnoughSpaceAboveForBig = windowHeight - bottomValue - contentMaxHeight > CONST.AUTO_COMPLETE_SUGGESTER.SUGGESTION_BOX_MAX_SAFE_DISTANCE;
-            const isEnoughSpaceAboveForSmall = windowHeight - bottomValue - contentMinHeight > CONST.AUTO_COMPLETE_SUGGESTER.SUGGESTION_BOX_MAX_SAFE_DISTANCE;
+            let bottomValue = windowHeight - (cursorCoordinates.y - scrollValue + y) - keyboardHeight;
```
[the remark this answers is not in the document]
Comment (perunt) [reply src/components/AutoCompleteSuggestions/index.tsx:85]: Yes, due to the coordinate grid starting from the top

**Review by rayane-d — APPROVED** [review]
LGTM

**Comment by perunt** [comment]
Thanks for the review!

Yes, I'm currently working on the second issue as a separate PR. However, the first issue looks odd, and I want to retest it.

@puneetlath, I have one more question about how the suggestion box should behave once it is no longer needed. Here's what I mean:
I just checked how it's implemented in Web Slack. Once the suggestion box is shown, the user can't scroll the input or the list. If the user taps on a suggestion item or outside the box, it disappears. So, I'm planning to mimic this behavior. What do you think?

To summarize, the suggestion box behavior would be as follows:
- The suggestion box opens.
- If the user taps outside the box (even on some touchable element like LHN or anything else), we close the suggestion box, and we do not count this touch as a click/touch/scroll.

So in that way, the user will be able just to close SB and not misclick anything else

**Comment by puneetlath** [comment]
@perunt your plan sounds good to me.

**Comment by perunt** [comment]
@rayane-djouah done

**Review by rayane-d — APPROVED** [review]
Thank you!

**Comment by perunt** [comment]
@puneetlath it looks good to go

**Comment by perunt** [comment]
Also, I just finished the task with hiding the suggestion box
https://github.com/Expensify/App/pull/45076

File: src/pages/home/report/ReportActionCompose/SuggestionEmoji.tsx [hunk 9]
```diff
@@ -193,6 +194,8 @@ function SuggestionEmoji(
 
     const getSuggestions = useCallback(() => suggestionValues.suggestedEmojis, [suggestionValues]);
 
+    const checkIfSuggestionVisible = useCallback(() => isEmojiSuggestionsMenuVisible, [isEmojiSuggestionsMenuVisible]);
```
Comment (puneetlath) [inline src/pages/home/report/ReportActionCompose/SuggestionEmoji.tsx:197]: Our guidelines for booleans are that they should start with `is` or `are` to indicate true/false. I think the following naming makes more sense for what this is doing based on my understanding. In that it will return true/false based on whether the suggestion menu is visible. 

```suggestion
    const isSuggestionsMenuVisible = useCallback(() => isEmojiSuggestionsMenuVisible, [isEmojiSuggestionsMenuVisible]);
```

File: src/pages/home/report/ReportActionCompose/SuggestionMention.tsx [hunk 10]
```diff
@@ -408,6 +408,7 @@ function SuggestionMention(
     );
 
     const getSuggestions = useCallback(() => suggestionValues.suggestedMentions, [suggestionValues]);
+    const checkIfSuggestionVisible = useCallback(() => isMentionSuggestionsMenuVisible, [isMentionSuggestionsMenuVisible]);
```
Comment (puneetlath) [inline src/pages/home/report/ReportActionCompose/SuggestionMention.tsx:411]:
```suggestion
    const isSuggestionsMenuVisible = useCallback(() => isMentionSuggestionsMenuVisible, [isMentionSuggestionsMenuVisible]);
```

File: src/pages/home/report/ReportActionCompose/Suggestions.tsx [hunk 11]
```diff
@@ -124,6 +124,11 @@ function Suggestions(
         suggestionEmojiRef.current?.setShouldBlockSuggestionCalc(shouldBlock);
         suggestionMentionRef.current?.setShouldBlockSuggestionCalc(shouldBlock);
     }, []);
+    const checkIfSuggestionVisible = useCallback((): boolean => {
```
Comment (puneetlath) [inline src/pages/home/report/ReportActionCompose/Suggestions.tsx:127]:
```suggestion
    const isSuggestionsMenuVisible = useCallback((): boolean => {
```

File: src/components/AutoCompleteSuggestions/index.tsx [hunk 12]
```diff
@@ -26,6 +26,14 @@ function isSuggestionRenderedAbove(isEnoughSpaceAboveForBig: boolean, isEnoughSp
     return isEnoughSpaceAboveForBig || isEnoughSpaceAboveForSmall;
 }
 
+type IsEnoughSpaceAbove = Pick<MeasureParentContainerAndCursor, 'y' | 'cursorCoordinates' | 'scrollValue'> & {
+    contentHeight: number;
+    topInset: number;
+};
+function isEnoughSpaceAbove({y, cursorCoordinates, scrollValue, contentHeight, topInset}: IsEnoughSpaceAbove): boolean {
```
Comment (puneetlath) [inline src/components/AutoCompleteSuggestions/index.tsx:33]: I find this naming slightly confusing. If I understand what this is doing correctly, then I think something like thisi s more clear: 
```suggestion
function isEnoughSpaceToRenderMenuAboveCursor({y, cursorCoordinates, scrollValue, contentHeight, topInset}: IsEnoughSpaceAbove): boolean {
```

File: src/pages/home/report/ReportActionCompose/ReportActionCompose.tsx [hunk 13]
```diff
@@ -60,6 +60,7 @@ type SuggestionsRef = {
     updateShouldShowSuggestionMenuToFalse: (shouldShowSuggestionMenu?: boolean) => void;
     setShouldBlockSuggestionCalc: (shouldBlock: boolean) => void;
     getSuggestions: () => Mention[] | Emoji[];
+    checkIfSuggestionVisible: () => boolean;
```
Comment (puneetlath) [inline src/pages/home/report/ReportActionCompose/ReportActionCompose.tsx:63]:
```suggestion
    isSuggestionsMenuVisible: () => boolean;
```

File: src/pages/home/report/ReportActionItemMessageEdit.tsx [hunk 14]
```diff
@@ -81,20 +89,25 @@ function ReportActionItemMessageEdit(
...
     const reportScrollManager = useReportScrollManager();
     const {translate, preferredLocale} = useLocalize();
     const {isKeyboardShown} = useKeyboardState();
     const {isSmallScreenWidth} = useWindowDimensions();
     const prevDraftMessage = usePrevious(draftMessage);
-
+    const suggestionsRef = useRef<SuggestionsRef>(null);
+    const mobileInputScrollPosition = useRef(0);
+    const cursorPositionValue = useSharedValue({x: 0, y: 0});
+    const tag = useSharedValue(-1);
```
Comment (puneetlath) [inline src/pages/home/report/ReportActionItemMessageEdit.tsx:101]: What is this used for? The name is very generic so I'm having a hard time understanding.

File: src/pages/home/report/ReportActionItemMessageEdit.tsx [hunk 15]
```diff
@@ -340,6 +374,17 @@ function ReportActionItemMessageEdit(
                 return;
             }
             const keyEvent = e as KeyboardEvent;
+            const isSuggestionActive = suggestionsRef.current?.checkIfSuggestionVisible();
```
Comment (puneetlath) [inline src/pages/home/report/ReportActionItemMessageEdit.tsx:377]:
```suggestion
            const isSuggestionsMenuVisible = suggestionsRef.current?.checkIfSuggestionVisible();
```

File: src/pages/home/report/ReportActionItemMessageEdit.tsx [hunk 16]
```diff
@@ -348,7 +393,57 @@ function ReportActionItemMessageEdit(
...
+                    height,
+                    scrollValue,
+                    cursorCoordinates: {x: xPosition, y: yPosition},
+                });
+            });
+        },
+        [cursorPositionValue.value, measureContainer, selection],
+    );
+
+    useEffect(() => {
```
Comment (puneetlath) [inline src/pages/home/report/ReportActionItemMessageEdit.tsx:428]: Could we document what the purpose of this effect is.

File: src/pages/home/report/ReportActionItemMessageEdit.tsx [hunk 17]
```diff
@@ -440,8 +547,24 @@ function ReportActionItemMessageEdit(
...
                         />
                     </View>
+
+                    <Suggestions
+                        ref={suggestionsRef}
+                        isComposerFocused={textInputRef.current?.isFocused()}
+                        updateComment={updateDraft}
+                        measureParentContainerAndReportCursor={measureParentContainerAndReportCursor}
+                        isGroupPolicyReport={false}
+                        // Input
```
Comment (puneetlath) [inline src/pages/home/report/ReportActionItemMessageEdit.tsx:561]: What's the purpose of this comment?

**Review by puneetlath — CHANGES_REQUESTED** [review]
Left a handful of comments, mostly about naming. Overall looks solid to me.

File: src/pages/home/report/ReportActionCompose/SuggestionEmoji.tsx (hunk 9 above)
Comment (perunt) [reply src/pages/home/report/ReportActionCompose/SuggestionEmoji.tsx:197]: Sure, it makes sense. The only concern is that it's a callback, and it would be quite confusing to do `isSuggestionsMenuVisible()`, so maybe it makes sense to name it `getIsSuggestionsMenuVisible`?

Comment (puneetlath) [reply src/pages/home/report/ReportActionCompose/SuggestionEmoji.tsx:197]: Ah cool. That makes sense to me.

File: src/pages/home/report/ReportActionCompose/SuggestionMention.tsx (hunk 10 above)
Comment (perunt) [reply src/pages/home/report/ReportActionCompose/SuggestionMention.tsx:411]: As was discussed above, I changed it to `getIsSuggestionsMenuVisible`

File: src/pages/home/report/ReportActionCompose/Suggestions.tsx (hunk 11 above)
Comment (perunt) [reply src/pages/home/report/ReportActionCompose/Suggestions.tsx:127]: As was discussed above, I changed it to `getIsSuggestionsMenuVisible`

File: src/components/AutoCompleteSuggestions/index.tsx (hunk 12 above)
Comment (perunt) [reply src/components/AutoCompleteSuggestions/index.tsx:33]: 👍🏻

File: src/pages/home/report/ReportActionItemMessageEdit.tsx (hunk 14 above)
Comment (perunt) [reply src/pages/home/report/ReportActionItemMessageEdit.tsx:101]: We use the tag to store the ID of the text input. Later, we use it in onSelectionChange to pick up the proper text input data. I left a comment

File: src/pages/home/report/ReportActionItemMessageEdit.tsx (hunk 16 above)
Comment (perunt) [reply src/pages/home/report/ReportActionItemMessageEdit.tsx:428]: Sure, I left a comment

File: src/pages/home/report/ReportActionItemMessageEdit.tsx (hunk 17 above)
Comment (perunt) [reply src/pages/home/report/ReportActionItemMessageEdit.tsx:561]: It appears after some refactoring in the suggestion component for the main composer, so I did the same for the edit input composer.
However, since we don't have guidelines for that, I'll remove it

**Comment by perunt** [comment]
@puneetlath can you take a look again?

**Review by puneetlath — APPROVED** [review]
